From 81d80ea8f650728e5b68c8ca65f76212db812d73 Mon Sep 17 00:00:00 2001
From: Ulf Hermann <ulf.hermann@qt.io>
Date: Wed, 20 Jan 2021 10:42:36 +0100
Subject: [PATCH 002/150] Partially revert
 813a928c7c3cf98670b6043149880ed5c955efb9

Otherwise ASAN complains about mismatched new/delete.

Change-Id: I76ddcc388309e20055c93c68b9b6fa8a11b884e1
Reviewed-by: Andrei Golubev <andrei.golubev@qt.io>
Reviewed-by: Ville Voutilainen <ville.voutilainen@qt.io>
(cherry picked from commit 606b03d0a35e71565c3a91588ef2db45b9262e27)
---
 src/corelib/tools/qsharedpointer_impl.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/corelib/tools/qsharedpointer_impl.h b/src/corelib/tools/qsharedpointer_impl.h
index 443e0d93d1..618b31e391 100644
--- a/src/corelib/tools/qsharedpointer_impl.h
+++ b/src/corelib/tools/qsharedpointer_impl.h
@@ -155,6 +155,9 @@ namespace QtSharedPointer {
 #endif
         inline void checkQObjectShared(...) { }
         inline void setQObjectShared(...) { }
+
+        inline void operator delete(void *ptr) { ::operator delete(ptr); }
+        inline void operator delete(void *, void *) { }
     };
     // sizeof(ExternalRefCountData) = 12 (32-bit) / 16 (64-bit)
 
-- 
2.40.0.rc2

